modules/tinymce/src/core/main/ts/keyboard/MatchKeys.ts

```diff
@@ -26,7 +32,17 @@ const defaultPatterns = (patterns: KeyPattern[]): KeyPattern[] => Arr.map(patter
   ...pattern
 }));
 
-const matchesEvent = (pattern: KeyPattern, evt: KeyboardEvent) => (
+const defaultDelayedPatterns = (patterns: KeyPatternDelayed[]): KeyPatternDelayed[] => Arr.map(patterns, (pattern) => ({
+  shiftKey: false,
+  altKey: false,
+  ctrlKey: false,
+  metaKey: false,
+  keyCode: 0,
+  action: Fun.noop,
```

Comment:
This doesn't match the type, so this shouldn't be compiling... looks like this should be this?
```suggestion
  action: () => Optional.some(Fun.noop),
```

Also there's a lot of duplication here with `defaultPatterns` and `defaultDelayedPatterns`, so can we please move that out into a global variable? e.g.
```
const baseKeyPattern = {
  shiftKey: false,
  altKey: false,
  ctrlKey: false,
  metaKey: false,
  keyCode: 0
};
```

- [ ] Remove duplication

Reply:
It seems like it doesn't typecheck at all in this one. Good catch with this mistake.
Also deduped as requested.